src/zapv2/__init__.py

```diff
@@ -110,25 +114,22 @@ def urlopen(self, *args, **kwargs):
            - `args`:  all non-keyword arguments.
            - `kwargs`: all other keyword arguments.
         """
-        kwargs['proxies'] = self.__proxies
-        return urllib.urlopen(*args, **kwargs).read()
+        return self.session.get(*args, proxies=self.__proxies, **kwargs).text
```

Comment:
So I think this is still a problem - it will add the api key to calls to the target, which we _really_ dont want to do. In fact we may need to rework the api to make sure we never do that...